ui/hooks/useAddressDetails.js

```diff
@@ -0,0 +1,48 @@
+import { useSelector } from 'react-redux';
+
+import { toChecksumHexAddress } from '../../shared/modules/hexstring-utils';
+import {
+  getAddressBook,
+  getMetaMaskIdentities,
+  getTokenList,
+  getUseTokenDetection,
+} from '../selectors';
+import { shortenAddress } from '../helpers/utils/util';
+
+const useAddressDetails = (toAddress) => {
+  const addressBook = useSelector(getAddressBook);
+  const identities = useSelector(getMetaMaskIdentities);
+  const tokenList = useSelector(getTokenList);
+  const useTokenDetection = useSelector(getUseTokenDetection);
+  const checksummedAddress = toChecksumHexAddress(toAddress);
+
+  if (!toAddress) {
+    return {};
+  }
+  const addressBookEntryObject = addressBook.find(
+    (entry) => entry.address === checksummedAddress,
+  );
+  if (addressBookEntryObject?.name) {
+    return { toName: addressBookEntryObject?.name, isTrusted: true };
+  }
+  if (identities[toAddress]?.name) {
+    return { toName: identities[toAddress]?.name, isTrusted: true };
```

Comment:
Since we know we have a `identities[toAddress]` at this point, we can skip the `?`.

Reply:
fixed 👍